src/cython_code/State.pyx

```diff
@@ -372,18 +372,34 @@ cdef class p_State:
     def insert_row(self, row_data, matching_row_idx, row_idx=-1):
         return self.thisptr.insert_row(row_data, matching_row_idx, row_idx)
     def transition(self, which_transitions=(), n_steps=1,
-                   c=(), r=(), max_iterations=-1, max_time=-1):
+                   c=(), r=(), max_iterations=-1, max_time=-1, progress=None):
+         def _progress(percentage):
+              import sys
+              progress = ' ' * 30
+              fill = int(percentage * len(progress))
+              progress = '[' + '=' * fill + progress[fill:] + ']'
+              print('\r{} {:1.2f}%'.format(progress, 100 * percentage), end="")
+              sys.stdout.flush()
```

Comment:
And thread it through the the `analyze` method of the Crosscat engine.  OK if it bottoms out inside cgpm for now, though it in the future should really be threaded through the cgpm `transition` method and bottom out in the bayeslite progress hook (tracked in <https://github.com/probcomp/bayeslite/issues/39>) so that applications using bayeslite can witness progress in all cgpm operations.